pkg/status/status.go

```diff
@@ -1,65 +1,89 @@
 package status
 
 import (
-	"context"
+	"fmt"
+	"sync"
+	"time"
 
 	configv1 "github.com/openshift/api/config/v1"
 	configclient "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
 	cohelpers "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
 	operatorhelpers "github.com/openshift/library-go/pkg/operator/v1helpers"
 	log "github.com/sirupsen/logrus"
-	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
-const clusterOperatorName = "marketplace-operator"
+const (
+	clusterOperatorName     = "marketplace-operator"
+	minSyncsBeforeReporting = 4
+	// At this point, a failure is critical so all syncs must be successful
+	// to report a working operator
+	successRatio           = 1.0
+	maxSyncsBeforeTruncate = 10000
+	syncTruncateValue      = 100
+)
+
+// SyncCh is used to report a sync event
+// The ratio of Successful Syncs / Total Syncs will be used to report operator health to the CVO
+var syncCh = make(chan error)
+
+// status will be a singleton
```

Comment:
I know this is a singleton, but it will be hard to test this if we don't compose it at all. Can these fields all be in the status type?

Reply:
syncCh will need to be a global variable because it's used by controllers but I will make sure that all other fields can be set in the call.

Reply:
@kevinrizza These fields have been moved into the status type.